main.tf

```
@@ -472,3 +472,105 @@ resource "aws_autoscaling_schedule" "this" {
   # Cron examples: https://crontab.guru/examples.html
   recurrence = lookup(each.value, "recurrence", null)
 }
+
+################################################################################
+# Target scaling policy schedule
+################################################################################
+resource "aws_autoscaling_policy" "this" {
+  for_each = var.create_asg && var.create_scaling_policy ? var.scaling_policies : {}
+
+  name                      = each.key
```

this looks great and I completely agree, but users can sometimes be thrown off by this. instead we can give them a little help by overloading or adding a fallback (support two ways)

```suggestion
  name                      = lookup(each.value, "name", each.key)
```

so they can specify the name in the map as per normal, but if not then we can fall back to use the map key